src/update.rs

```diff
@@ -263,3 +285,33 @@ fn api_headers(auth_token: &Option<String>) -> header::HeaderMap {
 
     headers
 }
+
+fn cleanup_temp_directories<P: AsRef<Path>>(
+    tmp_dir_parent: P,
+    tmp_dir_prefix: &str,
+    expected_tmp_filename: &str,
+) -> Result<()> {
+    for entry in fs::read_dir(tmp_dir_parent)? {
+        let entry = entry?;
+        let tmp_dir_name = if let Ok(tmp_dir_name) = entry.file_name().into_string() {
+            tmp_dir_name
+        } else {
+            continue;
+        };
+
+        // For safety, check that temporary directory contains only expected file before removing.
+        let is_expected_tmp_file = |tmp_file_entry: std::io::Result<fs::DirEntry>| {
+            tmp_file_entry
+                .ok()
+                .filter(|e| e.file_name() == expected_tmp_filename)
```

Comment:
do we care to check if the dir contains other subdirs (this is only checking the files in the dir, right?)?

Reply:
`fs::read_dir` returns subdirectories as well, so the cleanup fails if there are any. `self_update` doesn't create any subdirectories in the temp directory because it's only used for `replace_using_temp`. If subdirectories exist then the user is using the temp directory for something else (not sure why that would be the case, but wanted to be careful with `fs::remove_dir_all`).